mpt_api_client/models/model.py

```
@@ -113,6 +113,24 @@ def _process_item(self, item: Any) -> Any:
+    if target_class and isinstance(target_class, type) and issubclass(target_class, BaseModel):
+        model_class: type[BaseModel] = target_class
+        return model_class(**value)
+    return BaseModel(**value)
+
+
+def _resolve_list_model_class(target_class: Any) -> type["BaseModel"]:
+    """Resolves the element model class for a list-typed field from its type hint."""
+    if not target_class or get_origin(target_class) is not list:
+        return BaseModel
+    args = get_args(target_class)
+    if args and isinstance(args[0], type) and issubclass(args[0], BaseModel):  # noqa: WPS221
```

Ruff (0.15.15)

[warning] Invalid rule code in `# noqa`: WPS221

Add non-Ruff rule codes to the `lint.external` configuration option

(RUF102)